core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataCategory.java

```diff
@@ -0,0 +1,138 @@
+package org.fao.geonet.kernel.datamanager.base;
+
+import java.util.Collection;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+import org.fao.geonet.ApplicationContextHolder;
+import org.fao.geonet.domain.Metadata;
+import org.fao.geonet.domain.MetadataCategory;
+import org.fao.geonet.kernel.SvnManager;
+import org.fao.geonet.kernel.datamanager.IMetadataCategory;
+import org.fao.geonet.repository.MetadataCategoryRepository;
+import org.fao.geonet.repository.MetadataRepository;
+import org.fao.geonet.repository.Updater;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import jeeves.server.context.ServiceContext;
+
+public class BaseMetadataCategory implements IMetadataCategory {
+
+    @Autowired
+    private MetadataRepository metadataRepository;
+    @Autowired(required=false)
+    private SvnManager svnManager;
+    @Autowired
+    private ApplicationContext _applicationContext;
+
+    public void init(ServiceContext context, Boolean force) throws Exception {
+//        this.metadataRepository = context.getBean(MetadataRepository.class);
```

Comment:
I don't think `@Autowired` will work due to multinode. Please this class and other similar ones to use instead `context.getBean` again